src/Config/RectorConfig.php

```diff
@@ -75,11 +75,7 @@ public function disableParallel(): void
         SimpleParameterProvider::setParameter(Option::PARALLEL, false);
     }
 
-    /**
-     * Defaults in sync with https://phpstan.org/config-reference#parallel-processing
-     * as we run PHPStan as well
-     */
-    public function parallel(int $processTimeout = 120, int $maxNumberOfProcess = 32, int $jobSize = 20): void
+    public function parallel(int $processTimeout = 120, int $maxNumberOfProcess = 16, int $jobSize = 16): void
```

Comment:
@TomasVotruba I updated jobSize to 16 per your request on chat ;)

Reply:
Thanks. How is the performance on your projects? :)

Reply:
Tested on CodeIgniter 4 on 4.5 branch, it seems working ok

https://github.com/codeigniter4/CodeIgniter4/actions/runs/7540420845/job/20524948567?pr=8418

Reply:
@samsonasik Could you share a `composer.json` setup for testing this PR? I'd like community to test this, before we go for merge.

Reply:
It actually just `CodeIgniter4` project, switch to 4.5 branch:

```
git clone git@github.com:codeigniter4/CodeIgniter4.git
git checkut 4.5
```

The composer.json is as follow:

```
{
    "name": "codeigniter4/codeigniter4",
    "description": "The CodeIgniter framework v4",
    "license": "MIT",
    "type": "project",
    "homepage": "https://codeigniter.com",
    "support": {
        "forum": "https://forum.codeigniter.com/",
        "source": "https://github.com/codeigniter4/CodeIgniter4",
        "slack": "https://codeigniterchat.slack.com"
    },
    "require": {
        "php": "^8.1",
        "ext-intl": "*",
        "ext-mbstring": "*",
        "laminas/laminas-escaper": "^2.13",
        "psr/log": "^3.0"
    },
    "require-dev": {
        "codeigniter/coding-standard": "^1.7",
        "codeigniter/phpstan-codeigniter": "^1.4",
        "ergebnis/composer-normalize": "^2.28",
        "fakerphp/faker": "^1.9",
        "friendsofphp/php-cs-fixer": "~3.46.0",
        "kint-php/kint": "^5.0.4",
        "mikey179/vfsstream": "^1.6",
        "nexusphp/cs-config": "^3.6",
        "nexusphp/tachycardia": "^1.0",
        "phpstan/extension-installer": "^1.3",
        "phpstan/phpstan": "^1.10.2",
        "phpstan/phpstan-strict-rules": "^1.5",
        "phpunit/phpcov": "^8.2",
        "phpunit/phpunit": "^9.1",
        "predis/predis": "^1.1 || ^2.0",
        "rector/rector": "0.19.0",
        "vimeo/psalm": "^5.0"
    },
    "replace": {
        "codeigniter4/framework": "self.version"
    },
    "suggest": {
        "ext-curl": "If you use CURLRequest class",
        "ext-dom": "If you use TestResponse",
        "ext-exif": "If you run Image class tests",
        "ext-fileinfo": "Improves mime type detection for files",
        "ext-gd": "If you use Image class GDHandler",
        "ext-imagick": "If you use Image class ImageMagickHandler",
        "ext-libxml": "If you use TestResponse",
        "ext-memcache": "If you use Cache class MemcachedHandler with Memcache",
        "ext-memcached": "If you use Cache class MemcachedHandler with Memcached",
        "ext-mysqli": "If you use MySQL",
        "ext-oci8": "If you use Oracle Database",
        "ext-pgsql": "If you use PostgreSQL",
        "ext-readline": "Improves CLI::input() usability",
        "ext-redis": "If you use Cache class RedisHandler",
        "ext-simplexml": "If you format XML",
        "ext-sodium": "If you use Encryption SodiumHandler",
        "ext-sqlite3": "If you use SQLite3",
        "ext-sqlsrv": "If you use SQL Server",
        "ext-xdebug": "If you use CIUnitTestCase::assertHeaderEmitted()"
    },
    "autoload": {
        "psr-4": {
            "CodeIgniter\\": "system/",
            "Config\\": "app/Config/"
        },
        "exclude-from-classmap": [
            "**/Database/Migrations/**"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "CodeIgniter\\": "tests/system/",
            "Utils\\": "utils/"
        }
    },
    "config": {
        "allow-plugins": {
            "ergebnis/composer-normalize": true,
            "phpstan/extension-installer": true
        },
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "extra": {
        "branch-alias": {
            "dev-develop": "4.x-dev"
        }
    },
    "scripts": {
        "post-update-cmd": [
            "CodeIgniter\\ComposerScripts::postUpdate"
        ],
        "analyze": [
            "Composer\\Config::disableProcessTimeout",
            "bash -c \"XDEBUG_MODE=off phpstan analyse\"",
            "rector process --dry-run"
        ],
        "cs": [
            "Composer\\Config::disableProcessTimeout",
            "php-cs-fixer fix --ansi --verbose --dry-run --diff --config=.php-cs-fixer.user-guide.php",
            "php-cs-fixer fix --ansi --verbose --dry-run --diff --config=.php-cs-fixer.no-header.php",
            "php-cs-fixer fix --ansi --verbose --dry-run --diff"
        ],
        "cs-fix": [
            "Composer\\Config::disableProcessTimeout",
            "php-cs-fixer fix --ansi --verbose --diff --config=.php-cs-fixer.user-guide.php",
            "php-cs-fixer fix --ansi --verbose --diff --config=.php-cs-fixer.no-header.php",
            "php-cs-fixer fix --ansi --verbose --diff"
        ],
        "sa": "@analyze",
        "style": "@cs-fix",
        "test": "phpunit"
    },
    "scripts-descriptions": {
        "analyze": "Run static analysis",
        "cs": "Check the coding style",
        "cs-fix": "Fix the coding style",
        "test": "Run unit tests"
    }
}
```

@kenjis test in his PR:

- https://github.com/codeigniter4/CodeIgniter4/pull/8417#issuecomment-1893120212

for old config, even with 120 seconds timeout, but 20 job size.

Reply:
@samsonasik I meant how to install this branch commit into any project. So other can try in their projects.

Reply:
The steps can be:

```
composer config minimum-stability dev
composer config prefer-stable true
composer require rector/rector:dev-main --dev --ansi
```

Then update parallel config:

```php
$rectorConfig->parallel(120, 16,16);